### DISCUSSION TITLE
append depends on the number selected from dropdown

### DISCUSSION THREAD

**Discussion by nani1008** [react-hook-form/react-hook-form#5020] [Q&A]
So I have a dropdown which will let the user select a number after this it will append multiple form depending on the number selected. 

I want to append a form that depends on the number selected. 

Here is an example using codesandbox: 
https://codesandbox.io/s/delicate-pond-i8ksy?file=/src/App.js

**Comment by nani1008** [comment]
@bluebill1049 is this doable or not? Thank you! Please help.

**Reply by bluebill1049** [reply]
I think you should just attach a `onClick` on the `select` and then call `append([...])`

**Reply by bluebill1049** [reply]
<img width="933" alt="Screen Shot 2021-05-04 at 8 17 44 pm" src="https://user-images.githubusercontent.com/10513364/116989960-cb337e00-ad15-11eb-98dd-9463f9d6c1fa.png">

it supports array of value.

**Reply by bluebill1049** [reply]
Give it a try yourself first by reading the doc. That's the way to learn.

**Reply by nani1008** [reply]
Got it by setValue of my fieldArray to my slicedArray(0, "selected number").

**Reply by bluebill1049** [reply]
I was referred to the following

```ts
onClick={() => {
  const total = getValues('yourInput'); // so it will return 1/2/3
  apend(new Array(parseInt(total)).map(() => ({ 
    fieldName: 'test',
  }))) // so it will append x amount of inputs
}}
```
